Bottom: ea51a4a7b5316c5fc375408a8cacaecdaf79313b
Top:    6902b45fea4e337dfe6e936cd7a8b9291b55f745
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 16:43:56 +0000

Refresh of copy-c-add-mask_flags-paramete

---

diff --git a/src/buffer.c b/src/buffer.c
index 53c15f5..44b181c 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -214,26 +214,7 @@ AddElementToBuffer (ElementType *Element)
   ElementType *element;
 
   element = GetElementMemory (Dest);
-  CopyElementLowLevel (Dest, element, Element, false, 0, 0, FOUNDFLAG);
-  CLEAR_FLAG (ExtraFlag, element);
-  if (ExtraFlag)
-    {
-      ELEMENTTEXT_LOOP (element);
-      {
-	CLEAR_FLAG (ExtraFlag, text);
-      }
-      END_LOOP;
-      PIN_LOOP (element);
-      {
-	CLEAR_FLAG (ExtraFlag, pin);
-      }
-      END_LOOP;
-      PAD_LOOP (element);
-      {
-	CLEAR_FLAG (ExtraFlag, pad);
-      }
-      END_LOOP;
-    }
+  CopyElementLowLevel (Dest, element, Element, false, 0, 0, FOUNDFLAG | ExtraFlag);
   return (element);
 }
